tests/worker/test_reworker.py

```diff
@@ -17,85 +19,158 @@
+    def hang_until_stop() -> MsgGenerator:
+        yield from bps.wait_for([stop_plan.wait])
+
     ctx = BlueskyContext()
     ctx.with_config(EnvironmentConfig())
+    ctx.plan(hang_until_stop)
     return ctx
 
 
 @pytest.fixture
-def worker(context: BlueskyContext) -> Iterable[Worker[Task]]:
-    worker = RunEngineWorker(context)
-    yield worker
-    worker.stop()
+def inert_worker(context: BlueskyContext) -> Worker[Task]:
+    return RunEngineWorker(context, stop_timeout=2.0)
+
+
+@pytest.fixture
+def worker(inert_worker: Worker[Task]) -> Iterable[Worker[Task]]:
+    inert_worker.start()
+    yield inert_worker
+    inert_worker.stop()
 
 
-def test_stop_doesnt_hang(worker: Worker) -> None:
-    worker.start()
+def test_stop_doesnt_hang(inert_worker: Worker) -> None:
+    inert_worker.start()
+    inert_worker.stop()
 
 
-def test_stop_is_idempontent_if_worker_not_started(worker: Worker) -> None:
-    ...
+def test_stop_is_idempontent_if_worker_not_started(inert_worker: Worker) -> None:
+    inert_worker.stop()
 
 
-def test_multi_stop(worker: Worker) -> None:
-    worker.start()
-    worker.stop()
+def test_multi_stop(inert_worker: Worker) -> None:
+    inert_worker.start()
+    inert_worker.stop()
+    inert_worker.stop()
 
 
-def test_multi_start(worker: Worker) -> None:
-    worker.start()
+def test_multi_start(inert_worker: Worker) -> None:
+    inert_worker.start()
     with pytest.raises(Exception):
-        worker.start()
-
-
-def test_runs_plan(worker: Worker) -> None:
-    assert_run_produces_worker_events(
-        [
-            WorkerEvent(
-                state=WorkerState.RUNNING,
-                task_status=TaskStatus(
-                    task_name="test", task_complete=False, task_failed=False
-                ),
-                errors=[],
-                warnings=[],
+        inert_worker.start()
+    inert_worker.stop()
+
+
+def test_submit_task(worker: Worker) -> None:
+    assert worker.get_pending_tasks() == []
+    worker.submit_task(_SIMPLE_TASK)
+    assert worker.get_pending_tasks() == [_SIMPLE_TASK]
+
+
+def test_submit_multiple_tasks(worker: Worker) -> None:
+    assert worker.get_pending_tasks() == []
+    worker.submit_task(_SIMPLE_TASK)
+    assert worker.get_pending_tasks() == [_SIMPLE_TASK]
+    worker.submit_task(_LONG_TASK)
+    assert worker.get_pending_tasks() == [_SIMPLE_TASK, _LONG_TASK]
+
+
+def test_stop_with_task_pending(inert_worker: Worker) -> None:
+    inert_worker.start()
+    inert_worker.submit_task(_SIMPLE_TASK)
+    inert_worker.stop()
+
+
+def test_clear_task(worker: Worker) -> None:
+    task_id = worker.submit_task(_SIMPLE_TASK)
+    assert worker.get_pending_tasks() == [_SIMPLE_TASK]
+    worker.clear_task(task_id)
+    assert worker.get_pending_tasks() == []
+
+
+def test_clear_nonexistant_task(worker: Worker) -> None:
+    with pytest.raises(KeyError):
+        worker.clear_task("foo")
+
+
+@pytest.mark.parametrize("num_runs", [2, 3, 4])
```

Comment:
What is gained from running this multiple numbers of tasks? Won't they all fail on the second one?

Reply:
This was a holdover from when I was experimenting with stacks, good spot, have changed.